### ISSUE TITLE
psm3 provider duplicates standard linux library function names

### ISSUE THREAD

**Issue by mwheinz** [ofiwg/libfabric#6798]
We're running into a problem with some development work we're doing with OFI because the PSM3 provider exports functions that begin "uuid_*". For example, uuid_parse() and uuid_unparse(). This causes link failures when we try to include the standard libuuid:

```
1745 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-pack.o): In function `uuid_pack':
1746 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/pack.c:39: multiple definition of `uuid_pack'
1747 ./.libs/libpsm3.a(libuuid_la-pack.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/pack.c:39: first defined here
1748 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-parse.o): In function `uuid_parse':
1749 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/parse.c:42: multiple definition of `uuid_parse'
1750 ./.libs/libpsm3.a(libuuid_la-parse.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/parse.c:42: first defined here
1751 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-psm_uuid.o): In function `__psm2_uuid_generate':
1752 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/psm_uuid.c:77: multiple definition of `__psm2_uuid_generate'
1753 ./.libs/libpsm3.a(libuuid_la-psm_uuid.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/psm_uuid.c:77: first defined here
1754 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-psm_uuid.o): In function `psmi_uuid_unparse':
1755 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/psm_uuid.c:105: multiple definition of `psmi_uuid_unparse'
1756 ./.libs/libpsm3.a(libuuid_la-psm_uuid.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/psm_uuid.c:105: first defined here
1757 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-psm_uuid.o): In function `psmi_uuid_parse':
1758 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/psm_uuid.c:111: multiple definition of `psmi_uuid_parse'
1759 ./.libs/libpsm3.a(libuuid_la-psm_uuid.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/psm_uuid.c:111: first defined here
1760 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-unpack.o): In function `uuid_unpack':
1761 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/unpack.c:39: multiple definition of `uuid_unpack'
1762 ./.libs/libpsm3.a(libuuid_la-unpack.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/unpack.c:39: first defined here
1763 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-unparse.o): In function `uuid_unparse_lower':
1764 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/unparse.c:63: multiple definition of `uuid_unparse_lower'
1765 ./.libs/libpsm3.a(libuuid_la-unparse.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/unparse.c:63: first defined here
1766 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-unparse.o): In function `uuid_unparse_upper':
1767 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/unparse.c:68: multiple definition of `uuid_unparse_upper'
1768 ./.libs/libpsm3.a(libuuid_la-unparse.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/unparse.c:68: first defined here
1769 /home/mheinz/local/libfabric-internal/.libs/libuuid.a(libuuid_la-unparse.o): In function `uuid_unparse':
1770 /home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/unparse.c:73: multiple definition of `uuid_unparse'
1771 ./.libs/libpsm3.a(libuuid_la-unparse.o):/home/mheinz/local/libfabric-internal/prov/psm3/psm3/libuuid/unparse.c:73: first defined here
```

**Comment by acgoldma** [comment]
I actually already did this internally, will show up in a future update to psm3 provider. Will leave this open to track.

**Comment by mwheinz** [comment]
> I actually already did this internally, will show up in a future update to psm3 provider. Will leave this open to track.

Thanks.

**Comment by shefty** [comment]
Is there a reason why the fix can't be pushed now?

**Comment by acgoldma** [comment]
This was fixed as part of #6877
